src/pfe/file-watcher/server/test/functional-test/suites/generic/tests/logging.ts

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import mocha from "mocha";
+import { expect } from "chai";
+import * as _ from "lodash";
+
+import * as genericLib from "../../../lib/generic";
+import { fail } from "assert";
+
+export default class LoggingTest {
+    testName: string;
+
+    constructor(testName: string) {
+        this.testName = testName;
+    }
+
+    run(runOnly?: boolean): void {
+        (runOnly ? describe.only : describe)(this.testName, () => {
+            const logLevels = ["info", "warn", "error", "trace", "debug", "invalid"];
+            const invalidLevel = _.cloneDeep(logLevels).pop();
+            for (const level of logLevels) {
+                it(`set logging level to ${level}`, async () => {
+                    try {
+                        await genericLib.setLoggingLevel(level);
```

Comment:
I notice one issue when I run the test cases, we don't have the PFE log for setting the log level to `warn` and `error`. If the test cases failed and we don't have PFE log, then it's not good for us to debug.

![Screen Shot 2019-08-12 at 2 23 50 PM](https://user-images.githubusercontent.com/15131537/62888243-dad28e00-bd0c-11e9-896f-7235b00b3fce.png)

Reply:
That is probably product related. We need to add the PFE log to the server code when level `warn` or `error` is selected.

Reply:
![Screen Shot 2019-08-12 at 5 42 59 PM](https://user-images.githubusercontent.com/15131537/62901053-c781eb80-bd29-11e9-816c-db33ac72ed96.png)

Does our product code treat this differently for different log level?

Reply:
Yeah, I don't know why it does not print the same for trace with `warn` or `error`

Reply:
It's fine, I don't think it's test case issue, given it's just set log level function, we can keep an eye on this later on